addons/mail/models/mail_channel_rtc_session.py

```diff
@@ -95,19 +95,23 @@ def _notify_peers(self, notifications):
                 payload_by_target[target]['notifications'].append(content)
         return self.env['bus.bus']._sendmany([(target, 'mail.channel.rtc.session/peer_notification', payload) for target, payload in payload_by_target.items()])
 
-    def _mail_rtc_session_format(self, complete_info=True):
+    def _mail_rtc_session_format(self, fields=None):
         self.ensure_one()
-        vals = {
-            'id': self.id,
-            'channelMember': self.channel_member_id.mail_channel_member_format()[0],
-        }
-        if complete_info:
-            vals.update({
-                'isCameraOn': self.is_camera_on,
-                'isDeaf': self.is_deaf,
-                'isSelfMuted': self.is_muted,
-                'isScreenSharingOn': self.is_screen_sharing_on,
-            })
+        if not fields:
+            fields = {'id': True, 'channelMember': {'id': True, 'channel': {}, 'persona': {'partner': {'id', 'name', 'im_status'}, 'guest': {'id', 'name', 'im_status'}}}, 'isCameraOn': True, 'isDeaf': True, 'isSelfMuted': True, 'isScreenSharingOn': True}
```

Comment:
I m not a fan, can't we just directly return an object with all the correct value assigned directly ? I don't understand the purpose.